tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/CodeFixVerifier.cs

```diff
@@ -0,0 +1,127 @@
+﻿namespace FakeItEasy.Analyzer.Tests.Helpers
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using FluentAssertions;
+    using FluentAssertions.Execution;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using Microsoft.CodeAnalysis.Formatting;
+
+    /// <summary>
+    /// Superclass of all Unit tests made for diagnostics with codefixes.
+    /// Contains methods used to verify correctness of codefixes
+    /// </summary>
+    public abstract partial class CodeFixVerifier : DiagnosticVerifier
+    {
+#if CSHARP
+        /// <summary>
+        /// Called to test a C# codefix when applied on the inputted string as a source
+        /// </summary>
+        /// <param name="oldSource">A class in the form of a string before the CodeFix was applied to it</param>
+        /// <param name="newSource">A class in the form of a string after the CodeFix was applied to it</param>
+        /// <param name="codeFixIndex">Index determining which codefix to apply if there are multiple</param>
+        /// <param name="allowNewCompilerDiagnostics">A bool controlling whether or not the test will fail if the CodeFix introduces other warnings after being applied</param>

```

Comment:
The text of this comment would prompt me to supply a `true` value when I wanted the test to fail if a new warning appears, but this is not the case, right?